Tidy task page about using hostAliases (#21439)
kubernetes/website: content/en/docs/concepts/services-networking/add-entries-to-pod-etc-hosts-with-host-aliases.md

title: Adding entries to Pod /etc/hosts with HostAliases
content_template: templates/concept
weight: 60
min-kubernetes-server-version: 1.7
---

{{< toc >}}

{{% capture overview %}}
Adding entries to a Pod's `/etc/hosts` file provides Pod-level override of hostname resolution when DNS and other options are not applicable. You can add these custom entries with the HostAliases field in PodSpec.

Modification not using HostAliases is not suggested because the file is managed by the kubelet and can be overwritten on during Pod creation/restart.
{{% /capture %}}

{{% capture body %}}

## Default hosts file content

> Start an Nginx Pod which is assigned a Pod IP:

Reviewer: ```suggestion
Start an Nginx Pod and assign a Pod IP:
```

By default, the `hosts` file only includes IPv4 and IPv6 boilerplates like
`localhost` and its own hostname.

## Adding additional entries with hostAliases

In addition to the default boilerplate, you can add additional entries to the
`hosts` file.  
For example: to resolve `foo.local`, `bar.local` to `127.0.0.1` and `foo.remote`,
`bar.remote` to `10.1.2.3`, you can configure HostAliases for a Pod under
`.spec.hostAliases`:

{{< codenew file="service/networking/hostaliases-pod.yaml" >}}

> Yoyu can start a Pod with that configuration by running:

Reviewer: typo
```suggestion
You can start a Pod with that configuration by running:
```

```shell
kubectl apply -f https://k8s.io/examples/service/networking/hostaliases-pod.yaml
```

```
pod/hostaliases-pod created
```

Examine a Pod's details to see its IPv4 address and its status:

```shell
kubectl get pod --output=wide
```

```
NAME                           READY     STATUS      RESTARTS   AGE       IP              NODE
hostaliases-pod                0/1       Completed   0          6s        10.200.0.5      worker0
```

> The `hosts` file content looks like this:

Reviewer: Nit: there's another one of these ("would look") at L44.
